pkg/scheduler/plugins/predicates/predicates.go

```diff
@@ -279,7 +331,9 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 
 	ssn.AddPrePredicateFn(pp.Name(), func(task *api.TaskInfo) error {
 		// Check NodePorts
-		nodePortFilter.PreFilter(context.TODO(), state, task.Pod)
+		if predicate.nodePortEnable {
+			nodePortFilter.PreFilter(context.TODO(), state, task.Pod)
```

Comment:
After adding `nodePortEnable` to judge, is it better to transfer the definition of `nodePortFilter` from the front to here? 

There are also other variable definitions, such as: `nodeaffinity`, `taintToleration`, `podAffinityFilter`, etc.